src/cmark.h

```diff
@@ -592,6 +592,29 @@ int cmark_parser_has_partially_consumed_tab(cmark_parser *parser);
 CMARK_EXPORT
 int cmark_parser_get_last_line_length(cmark_parser *parser);
 
+/** Add a child to 'parent' during the parsing process.
+ *
+ * If 'parent' isn't the kind of node that can accept this child,
+ * this function will back up till it hits a node that can, closing
+ * blocks as appropriate.
+ */
+CMARK_EXPORT
+cmark_node*cmark_parser_add_child(cmark_parser *parser,
+                                  cmark_node *parent,
+                                  cmark_node_type block_type,
+                                  int start_column);
```

Comment:
No, I suggest you look at the implementation of `add_child` in `src/blocks.c` , this function finalizes open parent blocks, and creates the new block as open. This is really a parsing-specific method, and it cannot be replaced by `cmark_node_append_child`